scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -1642,5 +1683,190 @@
+
+(deftest test-progress-update-aggregator
+  (let [actual-progress-aggregator sched/progress-aggregator]
+    (with-redefs [sched/progress-aggregator
+                  (fn progress-aggregator
+                    [pending-progress-threshold instance-id->progress-state {:keys [instance-id response-chan] :as data}]
+                    (let [result (actual-progress-aggregator pending-progress-threshold instance-id->progress-state data)]
+                      (when response-chan
+                        (async/close! response-chan))
+                      result))]
+      (letfn [(send [progress-aggregator-chan message & {:keys [sync] :or {sync false}}]
+                (let [message (cond-> message
+                                      sync (assoc :response-chan (async/promise-chan)))
+                      put-result (async/>!! progress-aggregator-chan message)]
+                  (when sync
+                    (or (not put-result)
+                        (-> message :response-chan async/<!!)))))]
+
+        (testing "no progress to publish"
+          (let [pending-progress-threshold 10
+                progress-state-chan (async/chan 1)
+                progress-aggregator-chan (sched/progress-update-aggregator pending-progress-threshold progress-state-chan)]
+            (is (= {} (async/<!! progress-state-chan)))
+            (is (= {} (async/<!! progress-state-chan)))
+
+            (async/close! progress-state-chan)
+            (async/close! progress-aggregator-chan)
+            (poll-until #(nil? (async/<!! progress-state-chan)) 100 1000)))
+
+        (testing "basic progress publishing"
+          (let [pending-progress-threshold 10
+                progress-state-chan (async/chan)
+                progress-aggregator-chan (sched/progress-update-aggregator pending-progress-threshold progress-state-chan)]
+            (send progress-aggregator-chan {:instance-id "i1" :progress-message "i1.m1" :progress-percent 10})
+            (send progress-aggregator-chan {:instance-id "i2" :progress-message "i2.m1" :progress-percent 10})
+            (send progress-aggregator-chan {:instance-id "i3" :progress-message "i3.m1" :progress-percent 10})
+            (send progress-aggregator-chan {:instance-id "i2" :progress-message "i2.m2" :progress-percent 25})
+            (send progress-aggregator-chan {:instance-id "i1" :progress-message "i1.m2" :progress-percent 45} :sync true)
+            (is (= {"i1" {:progress-message "i1.m2" :progress-percent 45}
+                    "i2" {:progress-message "i2.m2", :progress-percent 25}
+                    "i3" {:progress-message "i3.m1", :progress-percent 10}}
+                   (async/<!! progress-state-chan)))
+            (is (= {} (async/<!! progress-state-chan)))
+
+            (async/close! progress-state-chan)
+            (async/close! progress-aggregator-chan)
+            (poll-until #(nil? (async/<!! progress-state-chan)) 100 1000)))
+
+        (testing "basic progress overflow on channel"
```

Comment:
Added an assertion to verify dropped messages by maintaining counter of processed messages